File: launchdarkly/python/src/PostFlagConfigScheduledChangesExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    post_flag_scheduled_changes_input = models.PostFlagScheduledChangesInput(
        executionDate=1718467200000,
        instructions=json.loads("""
            [
                {
                    "kind": "turnFlagOn"
                }
            ]
        """),
        comment="Optional comment describing the scheduled changes",
    )

    try:
        response = api.ScheduledChangesApi(api_client).post_flag_config_scheduled_changes(
            project_key="projectKey_string",
            feature_flag_key="featureFlagKey_string",
            environment_key="environmentKey_string",
            post_flag_scheduled_changes_input=post_flag_scheduled_changes_input,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling ScheduledChangesApi#post_flag_config_scheduled_changes: %s\n" % e)

File: launchdarkly/ruby/src/PostFlagConfigScheduledChangesExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["Authorization"] = "YOUR_API_KEY"
end

post_flag_scheduled_changes_input = LaunchDarklyClient::PostFlagScheduledChangesInput.new
post_flag_scheduled_changes_input.execution_date = 1718467200000
post_flag_scheduled_changes_input.instructions = JSON.parse(<<-EOD
    [
        {
            "kind": "turnFlagOn"
        }
    ]
    EOD
)
post_flag_scheduled_changes_input.comment = "Optional comment describing the scheduled changes"

begin
    response = LaunchDarklyClient::ScheduledChangesApi.new.post_flag_config_scheduled_changes(
        "projectKey_string", # project_key
        "featureFlagKey_string", # feature_flag_key
        "environmentKey_string", # environment_key
        post_flag_scheduled_changes_input,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling ScheduledChangesApi#post_flag_config_scheduled_changes: #{e}"
end
